[ch1935] [`version-release`] Add `npm release` control

hallais opened the pull request:
refactor: Add `npm release` control through `.releaseVersion({ npmPublish: true | false })`
docs: Add `README`

hallais committed:
[`release-version`] Adding `params.npmPublish` to manage package release [ch1935]

hallais committed:
[`package-lock.json`] Removing lock file

hallais committed:
[`README`] Adding doc

hallais committed:
[`index`] Adding padding

ghost commented:
## DeepCode's analysis on [#df1cc6](https://www.deepcode.ai/app/gh/contartec-team/version-release/692bc4a3d8aac1db171969352b5cdd5a98518cb4/contartec-team/version-release/df1cc6fc1348431a76ab6214877c2ae9465fdda1/pr/_/%2F/code/?utm_source=gh_review&c=0&w=2&i=3&) found:

- :warning: **2** warnings, :information_source: **3** minor issues. :point_down:
- :heavy_check_mark: **1** issue was fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">The function defined here takes 0 parameter(s), but is called here with 1 argument(s). The extra arguments will be ignored. Occurrences: <ul><li><a href="https://github.com/contartec-team/version-release/blob/df1cc6fc1348431a76ab6214877c2ae9465fdda1/docs/assets/site.js#L41">site.js:41</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/contartec-team/version-release/692bc4a3d8aac1db171969352b5cdd5a98518cb4/contartec-team/version-release/df1cc6fc1348431a76ab6214877c2ae9465fdda1/pr/_/%2Fdocs%2Fassets%2Fsite.js/javascript%2Fdc_interfile_project%2FWrongNumberOfArguments/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Calling map on a non-array object of type number (from number literal 100) will not work. Occurrences: <ul><li><a href="https://github.com/contartec-team/version-release/blob/df1cc6fc1348431a76ab6214877c2ae9465fdda1/docs/assets/split.js#L409">split.js:409</a></li> <li><a href="https://github.com/contartec-team/version-release/blob/df1cc6fc1348431a76ab6214877c2ae9465fdda1/docs/assets/split.js#L715">split.js:715</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/contartec-team/version-release/692bc4a3d8aac1db171969352b5cdd5a98518cb4/contartec-team/version-release/df1cc6fc1348431a76ab6214877c2ae9465fdda1/pr/_/%2Fdocs%2Fassets%2Fsplit.js/javascript%2Fdc_interfile_project%2FArrayMethodOnNonArray/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Unexpected 'this'. Occurrences: <ul><li><a href="https://github.com/contartec-team/version-release/blob/df1cc6fc1348431a76ab6214877c2ae9465fdda1/docs/assets/site.js#L29">site.js:29</a></li> <li><a href="https://github.com/contartec-team/version-release/blob/df1cc6fc1348431a76ab6214877c2ae9465fdda1/docs/assets/site.js#L55">site.js:55</a></li> <li><a href="https://github.com/contartec-team/version-release/blob/df1cc6fc1348431a76ab6214877c2ae9465fdda1/docs/assets/site.js#L56">site.js:56</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/contartec-team/version-release/692bc4a3d8aac1db171969352b5cdd5a98518cb4/contartec-team/version-release/df1cc6fc1348431a76ab6214877c2ae9465fdda1/pr/_/%2Fdocs%2Fassets%2Fsite.js/javascript%2Feslint%2Fno-invalid-this/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/contartec-team/version-release/692bc4a3d8aac1db171969352b5cdd5a98518cb4/contartec-team/version-release/df1cc6fc1348431a76ab6214877c2ae9465fdda1/pr/_/%2F/code/?utm_source=gh_review&c=0&w=2&i=3&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=contartec-team)

hallais committed:
[`release-version`]
Improving `doc`
Adding `release` object log
Adding `return` to `generateRelease`

hallais committed:
[`.npmrc`] Adding `npm` conf file

hallais committed:
[`.gitignore`] Adding `package-lock`

hallais committed:
[`package.json`] Adding `docs` and `update:docs` commands

hallais committed:
[`README`] Adding `Docs` section

hallais committed:
[`package.json`] Adding missing `husky` lib

hallais committed:
Updating `docs`

hallais committed:
[`release-version`] Improving `doc`

hallais committed:
[`package.json`] Updating libs

hallais committed:
[`release-version`] Readding `{ exec }` import

hallais committed:
[`package.json`] Adding `lint` command

hallais committed:
[`update-version`] Fixing lint errors

hallais committed:
[`package.json`] Refactoring `lint` command

hallais committed:
[`release-version`]
Adding `return` to `releaseVersion`
Fixing `jsdoc` lint errors

hallais committed:
[`update-version`] Fixing `jsdoc` lint errors

hallais committed:
[`package.json`] Adding `lint` to `pre-push` hook

hallais committed:
Updating `docs`
